### ISSUE TITLE
Possible encoding bug in handleCustomPacket()

### ISSUE THREAD

**Issue by pcbeard** [jython234/MagicClient#3]
I was going over your code carefully and noticed this:

```
            case MC_SERVER_HANDSHAKE:
                client.getLogger().debug("Server handshake!");
                ClientHandshake ch = new ClientHandshake((short) client.getServerPort());
                ch.encode();
                addToQueue(ch);
```

When I look at the implementation of ClientHandshake.encode(), I noticed that it encodes the first byte of the packet as the value MC_CLIENT_HANDSHAKE = (byte) 0x13. However, my reading of the MCPE client protocol document:

http://wiki.vg/Pocket_Minecraft_Protocol#Client_Handshake_.280x13.29

says that this packet is supposed to be "encoded using packet encapsulation format.." Your code is sending it unencoded, which might explain why your connections time out.

**Comment by pcbeard** [comment]
Ah, never mind. I see that when you call addToQueue(), you're really assembling RakNetInternalPacket   packets and wrapping them in a RakNetCustomPacket.

**Comment by pcbeard** [comment]
Closing. Apologies for the noise.

**Comment by alejzeis** [comment]
No problem, thanks for looking at my code! I just don't know why it's not
working.

On Tuesday, February 17, 2015, pcbeard notifications@github.com wrote:

> Closed #3 https://github.com/jython234/MagicClient/issues/3.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jython234/MagicClient/issues/3#event-237088156.

##

**Comment by pcbeard** [comment]
Using your code as a reference. I'm currently mystified about why MCPE sends 5 bytes more data in its OPEN_CONNECTION_REPLY_2 = 0x08 than PocketMine sends. I suppose it could just be the evolution of the protocol in 0.10.X.
